tools/make/golang.mk

```diff
@@ -32,8 +32,8 @@ go.build: $(addprefix go.build., $(addprefix $(PLATFORM)., $(BINS)))
 go.build.multiarch: $(foreach p,$(PLATFORMS),$(addprefix go.build., $(addprefix $(p)., $(BINS))))
 
 .PHONY: go.test.unit
-go.test.unit: ## Run go unit tests
-	go test ./...
+go.test.unit: $(tools/setup-envtest) ## Run go unit tests
```

Comment:
should these integration tests introduced by kubebuilder be separated into another make target and tagged separately (`// +build integration`)

Reply:
looks like a separate make target already exists here https://github.com/envoyproxy/gateway/blob/bab71d3a127ff5b05d2d8e2d217632e87bdbd595/tools/make/kube.mk#L11
rming the `setup-envtest` from here, using `kube-integration` build tags and editing the above `kube-test` target to `go test --tags=kube-integration ...` seems like a cleaner option imho